control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -462,15 +468,29 @@ bool AEB::checkCollision(MarkerArray & debug_markers)
     return objects;
   };
 
-  auto check_collision = [&](
-                           const Path & path, const colorTuple & debug_colors,
-                           const std::string & debug_ns,
-                           PointCloud::Ptr points_belonging_to_cluster_hulls) {
-    time_keeper_->start_track("has_collision_with_" + debug_ns);
-    auto objects =
-      get_objects_on_path(path, points_belonging_to_cluster_hulls, debug_colors, debug_ns);
-    // Get only the closest object and calculate its speed
+  auto check_collision = [&](const Path & path, std::vector<ObjectData> & objects) {
+    time_keeper_->start_track("has_collision");
     const auto closest_object_point = std::invoke([&]() -> std::optional<ObjectData> {
+      // Attempt to find the closest target object
+      const auto closest_target_object_itr =
+        std::min_element(objects.begin(), objects.end(), [](const auto & o1, const auto & o2) {
+          return (o1.is_target && (!o2.is_target || o1.distance_to_object < o2.distance_to_object));
+        });
```

Comment:
If o1 and o2 are NOT targets, doesn't that make o2 always the "closest" object regardless of distance? Maybe the logic should include all cases? 

```suggestion
//target objects have priority
if (o1.is_target != o2.is_target) {
    return o1.is_target;
}
return o1.distance_to_object < o2.distance_to_object;       
```

Reply:
I agree, updated :+1: